### PR TITLE
[Test] Convert transpose e8 test harness

### PR DESCRIPTION
**Pull request by Aaron-Hutchinson**
This converts the e8 transpose test harness to the new format added in #56. I limited the PR to e8 to simplify the review process, and will open another PR for converting e16+e32 once this one merges.

### CODE REVIEW

File: test/transpose/rvv/skl_transpose_e8_zve32x.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+#ifdef SKL_ENABLE_TESTS
+  // Verification tests
+  /* Edge cases */
+  {TEST, .m =   0, .n =   0},
+  {TEST, .m =   0, .n =   1},
+  {TEST, .m =   1, .n =   0},
+  {TEST, .m =   1, .n =   1},
+
+  /* e8m1 boundary*/
```
Comment (ericlove) [inline test/transpose/rvv/skl_transpose_e8_zve32x.c]: This depends on VLEN.

**Review by ericlove — COMMENTED** [review]
LGTM overall, just wish we could find a way to avoid hard-coding edge cases that are specific to one SiFive product in the test for a more general ISA...Any ideas?

File: test/transpose/rvv/skl_transpose_e8_zve32x.c (hunk 1 above)
Comment (Aaron-Hutchinson) [reply test/transpose/rvv/skl_transpose_e8_zve32x.c]: Hmm, good point. Maybe this should just test around powers of 2, ignoring any hardware properties or implementation details.

File: test/transpose/xsfmmbase/skl_transpose_e8_xsfmmbase.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  const transpose_e8_t *h = (transpose_e8_t *)t->harness;
+
+  skl_transpose_e8_xsfmmbase(h->m, h->n, h->a.data, h->rsa, h->at.data,
+                             h->rsat);
+}
+
+int main(void) {
+  // Set default row strides to row length
+  for (size_t i = 0; i < suite.num_tests; i++) {
+    tests[i].rsa = tests[i].rsa ? tests[i].rsa : tests[i].n;
```
Comment (myeh01) [inline test/transpose/xsfmmbase/skl_transpose_e8_xsfmmbase.c:106]: May be nice to add in checks that `rsa >= n` and `rsat >= m` in this and the rvv harness.

**Review by myeh01 — DISMISSED** [review]
LGTM

**Review by Harlen126 — DISMISSED** [review]
LGTM!

File: test/transpose/xsfmmbase/skl_transpose_e8_xsfmmbase.c (hunk 2 above)
Comment (Aaron-Hutchinson) [reply test/transpose/xsfmmbase/skl_transpose_e8_xsfmmbase.c:106]: I've just added `SKL_TEST_REQUIRE`s inside the harness init for this.

**Review by myeh01 — APPROVED** [review]
